pkgs/development/compilers/factor-lang/mk-factor-application.nix

```diff
@@ -0,0 +1,141 @@
+{
+  stdenv,
+  lib,
+  writeText,
+  makeWrapper,
+  factor-lang,
+  factor-no-gui,
+  librsvg,
+  gdk-pixbuf,
+}@initAttrs:
+
+drvArgs:
+
+let
+  flang = factor-lang; # workaround to satisfy nixf-tidy
+in
+(stdenv.mkDerivation drvArgs).overrideAttrs (
+  finalAttrs:
+  {
+    name ? "${finalAttrs.pname}-${finalAttrs.version}",
+    factor-lang ? if enableUI then flang else factor-no-gui,
+    enableUI ? false,
+    # Allow overriding the path to the deployed vocabulary name.  A
+    # $vocabName.factor file must exist!
+    vocabName ? finalAttrs.pname or name,
+    # Allow overriding the binary name
+    binName ? lib.last (lib.splitString "/" vocabName),
+    # Extra libraries needed
+    extraLibs ? [ ],
+    # Extra binaries in PATH
+    extraPaths ? [ ],
+    # Extra vocabularies needed by this application
+    extraVocabs ? [ ],
+    deployScriptText ? ''
+      USING: command-line io io.backend io.pathnames kernel namespaces sequences
+      tools.deploy tools.deploy.config tools.deploy.backend vocabs.loader ;
+
+      IN: deploy-me
+
+      : load-and-deploy ( path/vocab -- )
+          normalize-path [
+              parent-directory add-vocab-root
+          ] [
+              file-name dup reload deploy
+          ] bi ;
+
+      : deploy-vocab ( path/vocab path/target -- )
+          normalize-path deploy-directory set
+          f open-directory-after-deploy? set
+          load-and-deploy ;
+
+      : deploy-me ( -- )
+          command-line get dup length 2 = [
+              first2 deploy-vocab
+          ] [
+              drop
+              "usage: deploy-me <PATH-TO-VOCAB> <TARGET-DIR>" print
+              nl
+          ] if ;
+
+      MAIN: deploy-me
+    '',
+    ...
+  }@attrs:
+  let
+    deployScript = writeText "deploy-me.factor" finalAttrs.deployScriptText;
```

Comment:
How about using `passAsFile`?

Reply:
This does not work here. I am not entirely sure why but may have to do with defining the attrSet to `overrideAttrs` at this point and not the original `drvArgs`. It would be a nuissance to change that. So, I kept it as it is.